File: 02_Essential/Pointer_to_Structure/main.go
package main

import "fmt"

// taking a structure
type Rectangle struct {
	length, breadth int // taking variables
}

// Main Function
func main() {
	r := Rectangle{10, 5} // creating the instance of the struct type (создание экземпляра структурного типа)

	// Here, it is the pointer to the struct
	p := &r

	// updating the value of name
	p.length = 20
	p.breadth = 30

	Area := p.length * p.breadth

	fmt.Printf("%d/n", Area)
	return
}

File: 02_Essential/Pointer_to_Structure/main.cpp
// #include <stdio.h>
// struct Rectangle
// {
//     int length;
//     int breadth;
// };
// int main ()
// {
//     struct Rectangle r = {10, 5};
//     struct Rectangle *p = &r;
    
//     p -> length = 20;
//     p -> breadth = 30;
    
//     int Area = p -> length * p -> breadth;
//     printf("%d",Area);
// }

#include <stdio.h>
#include <stdlib.h>
struct Rectangle
{
    int length;
    int breadth;
};
int main ()
{
    struct Rectangle *p; // Create a pointer variable
    p=(struct Rectangle *) malloc(sizeof (struct Rectangle)); //Allocating Memory in Heap
    
    p -> length=10;
    p -> breadth=5;

    int Area = p -> length * p -> breadth;
    printf("%d", Area);
}